src/components/content/Content.tsx

```diff
@@ -1,21 +1,73 @@
-import React from "react";
-import { Button } from "antd";
-
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import React, { useEffect, useState } from "react";
 import "./style.scss";
-import Dino from "../dino/Dino";
+import {
+  loadInterestData,
+  loadContentData,
+  saveContentData,
+} from "../../utils/chatDataUtils";
+import { fetchGeminiNanoResponse } from "../../utils/fetchGeminiResponse";
+import { Message } from "../../types/messageType";
 
 const Content: React.FC = () => {
+  const [contentTags, setContentTags] = useState<string | null>(null);
+  const [contentResponse, setContentResponse] = useState<string | null>(null);
+  const [messages, setMessages] = useState<Message[]>([]);
+  const [loading, setLoading] = useState<boolean>(false);
+  const [aiLoading, setAiLoading] = useState<boolean>(false);
+
+  useEffect(() => {
+    setLoading(true);
+
+    loadInterestData((savedTags) => {
+      setContentTags(savedTags || null);
+      setLoading(false);
+    });
+
+    loadContentData((savedContent) => {
+      setContentResponse(savedContent || null);
+    });
+  }, []);
+
+  const generateContentTags = async () => {
+    const inputMessage = contentTags || "Generate content tags";
+    setAiLoading(true);
```

Comment:
Let's move this loading and fetcher bit directly to a `useFetchGeminiNanoResponse` kind of a custom hook, i'll do it in a seperate PR